### PR TITLE
fix(search): reset `query`, `page` and `sort` on clear query

### PR DESCRIPTION
**Pull request by mnavarroespinosa**
EX-7663

- Fixes issue that keeps the `page` param after clicking `Clear` query button in the search box. The same issue was happening with the `sort` parameter so I reset it too. 
- I opted for adding a new action `clearSearchParams` over using the existing `resetState` as the latter is only dispatched when requests are made and it's meant to evaluate the params that have changed between the old and new request. 
- Added an `e2e` regression test marked with `@regression` tag to facilitate identifying the regression tests.

### CODE REVIEW

**Comment by tiborux** [comment]
It looks nice, but I was trying to find a way to not "hardcoding" the parameters that we want to clean. Right now, we are using the resetStateWire but it needs the request to check if should we clean the parameters, so I don't know if someone can find a better way to clean these parameters, in the past, we did the same but cleaning one by one the parameters.

File: packages/x-components/src/x-modules/search/store/actions/clear-search-params.action.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { SearchXStoreModule } from '../types';
```
Comment (tajespasarela) [inline packages/x-components/src/x-modules/search/store/actions/clear-search-params.action.ts:1]: remove unnecessary action

**Review by tajespasarela — CHANGES_REQUESTED** [review]

**Review by diegopf — COMMENTED** [review]
This new approach seems that is not grouping changes properly when selecting filters now:

1. Scroll to page 2 (or beyond)
2. Select a filter
3. Two requests are made: one with the wrong page and another one with the page reset. The one with the wrong page is cancelled, but still we should avoid this.

File: packages/x-components/src/x-modules/search/events.types.ts [hunk 2]
```diff
@@ -39,9 +39,14 @@ export interface SearchXEvents {
   SearchRequestChanged: InternalSearchRequest | null;
   /**
    * Any property of the search request has been updated.
-   * Payload: The new internal request object.
+   * Payload: The new search request or `null` if there is not enough data in the state to
+   * conform a valid request.
+   *
+   * @remarks The difference from `SearchRequestChanged` and this event is this one will be executed
```
Comment (annacv) [inline packages/x-components/src/x-modules/search/events.types.ts:45]:
```suggestion
   * @remarks The difference between `SearchRequestChanged` and this event is this one will be executed
```

**Comment by annacv** [comment]
For me it's working fine

**Review by mnavarroespinosa — COMMENTED** [review]
Looks good to me!

**Review by tiborux — APPROVED** [review]

**Review by tajespasarela — APPROVED** [review]

**Pull request merged**